src/ui/projects/CreateScenePage.js

```diff
@@ -0,0 +1,366 @@
+  color: ${props => props.theme.red};
+  margin-top: 8px;
+`;
+
+export const UploadSceneSection = styled.form`
+  padding-bottom: 100px;
+  display: flex;
+
+  &:first-child {
+    padding-top: 100px;
+  }
+
+  h1 {
+    font-size: 36px;
+  }
+
+  h2 {
+    font-size: 16px;
+  }
+`;
+
+export const UploadSceneContainer = styled.div`
+  display: flex;
+  flex: 1;
+  flex-direction: column;
+  margin: 0 auto;
+  max-width: 800px;
+  min-width: 400px;
+  padding: 0 20px;
+`;
+
+export const SceneUploadHeader = styled.div`
+  margin-bottom: 36px;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`;
+
+const LeftContent = styled.div`
+  display: flex;
+  width: 360px;
+  border-top-left-radius: inherit;
+  align-items: flex-start;
+  padding: 30px;
+  position: relative;
+
+  img,
+  div {
+    width: 300px;
+    height: 168px;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    text-align: center;
+    background-color: ${props => props.theme.panel};
+    border-radius: 6px;
+  }
+  input {
+    opacity: 0;
+    position: absolute;
+  }
+`;
+
+const RightContent = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex: 1;
+  padding: 30px 30px;
+
+  label[type="button"] {
+    display: flex;
+    margin-bottom: 0;
+    margin-right: 5px;
+  }
+`;
+
+function CreateScenePage({ match, api }) {
+  const [isUploading, setIsUploading] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [sceneId, setSceneId] = useState(null);
+  const [error, setError] = useState(null);
+
+  const [sceneInfo, setSceneInfo] = useState({
+    name: "",
+    creatorAttribution: "",
+    allowRemixing: false,
+    allowPromotion: false
+  });
+
+  const [glbFile, setGlbFile] = useState(null);
+  const [thumbnailFile, setThumbnailFile] = useState(null);
+
+  const [thumbnailUrl, setThumbnailUrl] = useState(null);
+  const [sceneUrl, setSceneUrl] = useState(null);
+
+  const history = useHistory();
+
+  const openScene = useCallback(
+    function() {
+      window.open(sceneUrl);
+    },
+    [sceneUrl]
+  );
+
+  useEffect(
+    function() {
+      async function doInitialLoad() {
+        const sceneId = match.params.sceneId;
+        const isNew = sceneId === "new";
+
+        const scene = isNew ? {} : await api.getScene(sceneId);
+        setSceneInfo({
+          name: scene.name || "",
+          creatorAttribution: (scene.attributions && scene.attributions.creator) || "",
+          allowRemixing: scene.allow_remixing,
+          allowPromotion: scene.allow_promotion
+        });
+        setThumbnailUrl(scene.screenshot_url);
+        setSceneId(scene.scene_id);
+        setSceneUrl(scene.url);
+        setIsLoading(false);
+      }
+      doInitialLoad();
+    },
+    [match, api, setSceneInfo, setThumbnailUrl, setSceneId, setSceneUrl, setIsLoading]
+  );
+
+  const onChangeName = useCallback(
+    function(name) {
+      setSceneInfo({ ...sceneInfo, name });
```

Comment:
Rather than requiring `sceneInfo` as a dependency of this useCallback function, you can instead use `setSceneInfo(sceneInfo => ({ ...sceneInfo, name }))`.

More info on that form here: https://reactjs.org/docs/hooks-reference.html#functional-updates